From 6297c72245dcc0545efd058250d61fd73614bf50 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 11 Dec 2015 16:07:20 +0530
Subject: [PATCH 5/5] [EM/PMIC_CCSM]: Fix device mode detection issue after
 power role swap

Cable type cant be determined after power role swap. This leads to not
sending the otg events for cable disconnect. So, device mode cannot be
detected further. Hence allowing to send notifications to otg upon
cable disconnect even though cable type can't be determined.

Change-Id: I4e612913313676da8e22243c289928970a979f91
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-12469
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/448734
---
 drivers/power/intel_pmic_ccsm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/drivers/power/intel_pmic_ccsm.c
===================================================================
--- a/drivers/power/intel_pmic_ccsm.c	2016-04-20 13:02:34.090679318 +0800
+++ b/drivers/power/intel_pmic_ccsm.c	2016-04-20 13:03:34.982544781 +0800
@@ -866,7 +866,7 @@
 		time += USBSRCDET_SLEEP_RETRYDET;
 	} while (i < USBSRCDET_RETRY_CNT);
 
-	if ((val & USBSRCDET_SUSBHWDET_DETSUCC) !=
+	if ((val & USBSRCDET_SUSBHWDET) !=
 			USBSRCDET_SUSBHWDET_DETSUCC) {
 		dev_err(chc.dev, "Charger detection unsuccessful after %dms\n",
 			time);
